packages/fleet/lib/supervisor.integration.test.ts

```diff
@@ -0,0 +1,165 @@
+import { expect, describe, it, beforeEach, afterEach, vi } from 'vitest';
+import { Ok } from '@nangohq/utils';
+import { Supervisor } from './supervisor.js';
+import { getTestDbClient } from './db/helpers.test.js';
+import * as deployments from './models/deployments.js';
+import * as nodes from './models/nodes.js';
+import { generateCommitHash, createNodeWithAttributes } from './models/helpers.test.js';
+import type { Deployment } from './types.js';
+import { FleetError } from './utils/errors.js';
+
+const mockNodeProvider = {
+    start: vi.fn().mockResolvedValue(Ok(undefined)),
+    terminate: vi.fn().mockResolvedValue(Ok(undefined)),
+    mockClear: () => {
+        mockNodeProvider.start.mockClear();
+        mockNodeProvider.terminate.mockClear();
+    }
+};
+
+describe('Supervisor', () => {
+    const dbClient = getTestDbClient('supervisor');
+    const supervisor = new Supervisor({ dbClient, nodeProvider: mockNodeProvider });
+    let previousDeployment: Deployment;
+    let activeDeployment: Deployment;
+
+    beforeEach(async () => {
+        await dbClient.migrate();
+        previousDeployment = (await deployments.create(dbClient.db, generateCommitHash())).unwrap();
+        activeDeployment = (await deployments.create(dbClient.db, generateCommitHash())).unwrap();
+    });
+
+    afterEach(async () => {
+        await dbClient.clearDatabase();
+        mockNodeProvider.mockClear();
+    });
+
+    it('should not have more than one instance processing at a time', async () => {
+        const supervisor1 = new Supervisor({ dbClient, nodeProvider: mockNodeProvider });
+        const supervisor2 = new Supervisor({ dbClient, nodeProvider: mockNodeProvider });
+        supervisor1.start();
+        supervisor2.start();
+        await new Promise((resolve) => setTimeout(resolve, 100));
+        expect(true).toBe(true);
+        await supervisor1.stop();
+        await supervisor2.stop();
+    });
```

Comment:
Not entirely what this does; that stuff doed not throw?

Reply:
I was just preparing the test for my next PR. That should not be in there. Removing.